contrib/compile-tests.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# Checks intended compile failures.
+
+set -euo pipefail
+
+REPO_DIR=$(git rev-parse --show-toplevel)
+
+compile_check() {
+    local dir_path=$1
+    local expected_status=$2
+    local result=0
+    local output
+    # All crates in the specified subdirectory must compile or fail based on expected_status
+    for dir in "$dir_path"/*; do
+        if [ -d "$dir" ]; then
```

Comment:
I've just realized this check turns typos in the function invocation into silent passes which is not great. I'm not going to block the PR over it but would be really nice to address.